control/preimage/src/helpers.rs

```diff
@@ -0,0 +1,114 @@
+use subxt::{OnlineClient, PolkadotConfig};
+use codec::Encode;
+
+use crate::Context;
+
+use crate::bridge_hub_runtime::{
+    self,
+    runtime_types::bridge_hub_rococo_runtime::RuntimeCall as BridgeHubRuntimeCall,
+};
+
+use crate::relay_runtime::runtime_types::{
+    polkadot_runtime::RuntimeCall as RelayRuntimeCall,
+    sp_weights::weight_v2::Weight,
+    xcm::{
+        double_encoded::DoubleEncoded,
+        v2::OriginKind,
+        v3::{Instruction::{self, *}, junction::Junction, junctions::Junctions, WeightLimit, MaybeErrorCode, Xcm},
+        VersionedXcm,
+        VersionedMultiLocation
+    },
+    staging_xcm::v3::multilocation::MultiLocation,
+    pallet_xcm
+};
+
+use sp_arithmetic::helpers_128bit::multiply_by_rational_with_rounding;
+use sp_arithmetic::per_things::Rounding;
+
+// Increase call weight by 10% as a buffer in case the chain is upgraded with new weights
+// while the proposal is still in flight.
+pub fn increase_weight(ref_time: &mut u64, proof_size: &mut u64) {
+    let x = multiply_by_rational_with_rounding(*ref_time as u128, 110, 100, Rounding::Up).expect("overflow");
+    let y = multiply_by_rational_with_rounding(*proof_size as u128, 110, 100, Rounding::Up).expect("overflow");
+
+    *ref_time = x.try_into().expect("overflow");
+    *proof_size = y.try_into().expect("overflow");
+}
+
+// Base weight costs for Transact and ExpectTransactStatus
+const TRANSACT_INSTR_WEIGHT: u64 = 10_000_000;
```

Comment:
Could take https://github.com/polkadot-fellows/runtimes/blob/2ef38e75b43b2b91d160936843fa8b3e4ffbf5fc/system-parachains/bridge-hubs/bridge-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_generic.rs#L82
as reference.

Reply:
Also some weight should be reserved for the `unpaid_execution`
https://github.com/polkadot-fellows/runtimes/blob/2ef38e75b43b2b91d160936843fa8b3e4ffbf5fc/system-parachains/bridge-hubs/bridge-hub-kusama/src/weights/xcm/pallet_xcm_benchmarks_generic.rs#L337-L343

Reply:
Actually in the latest upload, I just configured `UnpaidExecution` to use unlimited weight.